### ISSUE TITLE
WCAG Reflow - select data source popup not visible

### ISSUE THREAD

**Issue by JonCole** [grafana/grafana#106037]
** Pre-Requisites: ** 
Open the web page / screen of the web application that is to be tested. 
Open Chrome/Edge dev Inspect tools. Keyboard shortcut "Ctrl+Shift+I". 
Ensure devtools are customized to Dock location: "undock into separate window". Minimize devtools so they are not obscuring the page. 
Zoom the browser window of the page to be tested to 200%. 
Adjust the browser window's height and width to the required viewport of 320px*256px. As long as dev tools remain open while resizing the viewport size will be displayed in the upper right corner of the browser so the size can be verified. 
Refresh the page. 

Alternatively: 
Change the OS display settings to a resolution of 1280x1024 at 100% dpi scaling. 
Open the web page/ screen of the web application that is to be tested. 
Zoom browser 400%. 
Set the browser to **full screen**. Keyboard shortcut "F11" key. 
If any browser sidebars are still visible (e.g., the Edge discover sidebar), these _must_ be closed. 

**Steps to reproduce**:
TAB to "Add visualization" button and hit ENTER key. "Select data source" popup will appear. 
Observe that After applying reflow the “select data source” pop-up is not visible properly: 

**Actual Result**:

After applying reflow the “select data source” pop-up is not visible properly. 

**Expected Result**

After applying reflow the “select data source” pop-up should be visible properly

Impact: User using reflow resolution will not be able to see the data present on “select data source” if it is not visible properly. 

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)

https://www.w3.org/WAI/WCAG21/Understanding/reflow

**Environment**:
- Grafana version: 11.5.1
- Data source type & version: Azure Monitor
- User OS & Browser: Windows, Edge or Chrome
- Others:

**Comment by LinglingTong** [comment]
Severity: Sev1

**Comment by JonCole** [comment]
https://github.com/user-attachments/assets/2af2eb0e-1fca-44cf-8c25-b713ead57dca

**Comment by JonCole** [comment]
This issue still remains even on the nightly container image you provided us for testing.  Attached is a video recording of the issue for your reference.  Note that in this case, the tester used the "Alternative" repro instructions provided previously (using 400% zoom)
https://github.com/user-attachments/assets/8e9ad9e9-5631-4e28-8781-87914d567f46

**Comment by ashharrison90** [comment]
@JonCole it sounds like testing is being done at 1280x720@400% zoom, but the ticket (and WCAG guidelines) specify 1280x1024@400% zoom. note the difference in vertical height. can you get confirmation that it's been tested at the correct size before we investigate further 🙏

**Comment by JonCole** [comment]
Yes, I will get clarification on the requirements.

**Comment by JonCole** [comment]
It looks like 1280 x 1024 at 400% is fine (does not have to be 720).  However, they said that they still see this issue reproduce even at 1024.  Can you check on your end?

<img width="1259" height="907" alt="Image" src="https://github.com/user-attachments/assets/89abd2e7-a479-4f60-9ccc-5fa6830f3780" />

**Comment by ashharrison90** [comment]
<img width="1282" height="1028" alt="Image" src="https://github.com/user-attachments/assets/e8f5635c-f6b7-4b99-836a-5480db187050" />

this is 1280x1024 @ 400%. dimensions are in the bottom left, x4 to get actual dimensions.

**Comment by joshhunt** [comment]
Hey @JonCole I note a _slight_ discrepancy with your alternate reproduction steps, and the WCAG reflow guidelines. Your report says:

> Change the **OS display settings** to a resolution of 1280x1024 at 100% dpi scaling.
> Open the web page/ screen of the web application that is to be tested.
> Zoom browser 400%.
> Set the browser to full screen. Keyboard shortcut "F11" key.

However the [WCAG Reflow guidelines](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) describe a browser _viewport_ instead of OS screen resolution:

> 320 CSS pixels is equivalent to a starting **viewport** width of 1280 CSS pixels wide at 400% zoom. For web content which is designed to scroll horizontally (e.g., with vertical text), 256 CSS pixels is equivalent to a starting viewport height of 1024 CSS pixels at 400% zoom.

Setting the browser to full screen should negate any OS chrome or toolbars from taking up space, but if the browser _window_ is set to 1280*1024, then browser chrome may take up space, making a viewport smaller than the WCAG guidelines.

When setting the viewport to 1280×1024 and zoom to 400% to get an _equivalent_ of 320×256, we can confirm the issue is fixed and all the UI is visible. Does this match with your expectations?

https://github.com/user-attachments/assets/3661c9fd-b6df-4b10-a169-5615947d93c3

**Comment by joshhunt** [comment]
@msftedad is still reproducing this in Chrome on Windows https://github.com/grafana/grafana/issues/112087

However, I'm still unable to reproduce the issue when sizing the browser window to 1280 x 1024 at a 400% zoom.

https://github.com/user-attachments/assets/5f5ce9fb-d41c-41fc-aec2-554eacd57dac

@msftedad @JonCole Here's me demoing the issue, showing that it's fixed. Can you spot a difference in our reproduction method that might explain this discrepancy?

**Comment by msftedad** [comment]
Hi @joshhunt, Issue is still repro'ing at our end. We followed below steps to apply Reflow settings.

**Prerequisites for Reflow for Web:** 

1. Set Display resolution to 1280*1024 in System settings.
2. Set Scaling to 100%.
3. Zoom Browser to 400%.

Please let us know if you need more information. Thanks

**Comment by ashharrison90** [comment]
> Hi [@joshhunt](https://github.com/joshhunt), Issue is still repro'ing at our end. We followed below steps to apply Reflow settings.
> 
> **Prerequisites for Reflow for Web:**
> 
> 1. Set Display resolution to 1280*1024 in System settings.
> 2. Set Scaling to 100%.
> 3. Zoom Browser to 400%.
> 
> Please let us know if you need more information. Thanks

@msftedad what are your browser dimensions in this case?

**Comment by msftedad** [comment]
@ashharrison90, We don’t modify any browser dimensions; they remain at their default values. We only adjust the display resolution and scaling in the system settings, and then zoom the browser to 400%.

**Comment by ashharrison90** [comment]
@msftedad are you setting the browser to be full screen? i think the WCAG guidelines are pretty clear in referring to viewport (e.g. browser) width/height, see https://www.w3.org/WAI/WCAG21/Understanding/reflow.html

see josh's comment [here](https://github.com/grafana/grafana/issues/106037#issuecomment-3355876189) for further info

if you're not setting the browser to full screen, there is browser chrome (e.g. tabs, url bar) taking up vertical space. please confirm the browser dimensions are 1280x1024 @ 400% zoom before attempting to reproduce - otherwise i don't think it's valid.

**Comment by JonCole** [comment]
@ashharrison90 I am still working with our accessibility test team on this, but I noticed we still seem to have a minor issue even in your screen recording

<img width="687" height="123" alt="Image" src="https://github.com/user-attachments/assets/f51052c6-b33d-444f-92cf-e0ff80bc272f" />

**Comment by JonCole** [comment]
Our test team also pointed out that even if the scrolling is working, the text is still getting clipped in the scrolled data:

<img width="1444" height="514" alt="Image" src="https://github.com/user-attachments/assets/a4eb1462-163d-489b-b011-59d8f68728a6" />

The above clip is from your screen recording.  I'm still trying to get a clear answer on whether the browser window resolution vs OS screen resolution issue.

**Comment by joshhunt** [comment]
Ideas discussed:
 - When the data source picker collapses into single-pane, we could switch to the normal modal scroll behaviour and use sticky header/footer for the search and buttons.
    - this could make it easier to wrap the DS description onto a second line
 - Adjust the padding at the smaller zoomed-in breakpoints
 - Wrap the "configure data source" text + button onto two lines
